### DISCUSSION TITLE
How to pass reportOptions on report request?

### DISCUSSION THREAD

**Discussion by smilekudo** [saleweaver/python-amazon-sp-api#572] [FAQ] [answered]
Hi,
I am using sp-api python to request report and it's worked but i don't know how to pass reportOptions to request.
Seem library not yet solve it?

```
cfg_report_type = 'GET_SALES_AND_TRAFFIC_REPORT'
    cfg_report_options = {"asinGranularity":"SKU"}
    # Create Report Request     
    #1#createReportResponse = x.create_report(reportType=cfg_report_type, dataStartTime=start_date_time, dataEndTime=end_date_time, reportOptions=cfg_report_options)
    createReportResponse = x.create_report(reportType=cfg_report_type, dataStartTime=start_date_time, dataEndTime=end_date_time, reportOptions='"reportOptions":{"asinGranularity":"SKU"}')
```

Reference: https://developer-docs.amazon.com/sp-api/docs/report-type-values#seller-retail-analytics-reports

Note: The dataStartTime value provided when requesting this report must not be more than two years before the date of the request. Failing this condition will result in the request being cancelled.

This report accepts the following reportOptions values:

dateGranularity - Data is available at different date range aggregation levels: DAY, WEEK, MONTH. Default: DAY.
asinGranularity - Data is available at different ASIN aggregation levels: PARENT, CHILD, SKU. Default: PARENT.
Example:
"reportOptions":{"dateGranularity":"MONTH", "asinGranularity":"CHILD"}

When dateGranularity is WEEK or MONTH the dataStartTime and dataEndTime values are expanded to that level of granularity. For WEEK the beginning is Sunday and the end is Saturday, and for MONTH the beginning is the first day of the month and the end is the last day of the month.

**Comment by saleweaver** [comment, marked as answer]
create_report(reportType=cfg_report_type, dataStartTime=start_date_time, dataEndTime=end_date_time, reportOptions=cfg_report_options)

Is exactly how, and is working as expected.
